### ISSUE TITLE
all_points_membership_vectors fails when no clusters are found

### ISSUE THREAD

**Issue by m-dz** [scikit-learn-contrib/hdbscan#157]
When no clusters are found (i.e. all points are outliers) the `all_points_membership_vectors` function fails with `ValueError: Invalid shape in axis 1: 0.` from lines:
https://github.com/scikit-learn-contrib/hdbscan/blob/98eef9987800d3f2dd790b0d2e6834ed365ef151/hdbscan/_prediction_utils.pyx#L276-L277

A quick and pretty efficient fix could be returning an array of zeros when `clusters.size == 0`:
```
    # Throws errors when no clusters, possible solution: return array of 0's
    if clusters.size == 0:
        return np.zeros(all_points.shape[0])
```
after those lines:
https://github.com/scikit-learn-contrib/hdbscan/blob/98eef9987800d3f2dd790b0d2e6834ed365ef151/hdbscan/prediction.py#L522-L524

What would you say?

Edit: Please ignore the commit referenced below, it's a bit WIP.

**Comment by lmcinnes** [comment]
Sounds good to me. PRs are welcome!

**Comment by m-dz** [comment]
Fixed by #159
